src/HealthGPS.Input/model_parser.cpp

```diff
@@ -24,29 +24,30 @@
 
 namespace hgps::input {
 
-hgps::RiskFactorSexAgeTable load_risk_factor_expected(const Configuration &config) {
+std::unique_ptr<hgps::RiskFactorSexAgeTable>
+load_risk_factor_expected(const Configuration &config) {
     MEASURE_FUNCTION();
 
     const auto &info = config.modelling.baseline_adjustment;
     if (!hgps::core::case_insensitive::equals(info.format, "CSV")) {
         throw hgps::core::HgpsException{"Unsupported file format: " + info.format};
     }
 
-    auto table = hgps::RiskFactorSexAgeTable{};
+    auto table = std::make_unique<hgps::RiskFactorSexAgeTable>();
```

Comment:
Why did you change `table` to a `std::unique_ptr` btw? Not a problem; just curious.

Reply:
I noted clang-tidy began complaining about model members as const and reference complicating automatic ctor generation -- see https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/avoid-const-or-ref-data-members.htmlv and links therein.

I suppose I could have made it into a shared pointer within the model class, but thought this way we only std move a pointer to the definition ctor. :shrug:

Reply:
Ah ok. That's interesting.

I think it's the right decision to return it as a `std::unique_ptr`. Out of interest, is it actually shared though? Or could it be left as a `std::unique_ptr`?

Reply:
It's shared once it enters the risk factor model definition class (with the risk factor model class). I keep it unique here and move it into a shared pointer inside the definition class. Not sure if that's good practice or not..